sdks/python/apache_beam/ml/inference/pytorch_inference.py

```diff
@@ -234,16 +266,17 @@ def run_inference(
     # If elements in `batch` are provided as a dictionaries from key to Tensors,
     # then iterate through the batch list, and group Tensors to the same key
     key_to_tensor_list = defaultdict(list)
-    for example in batch:
-      for key, tensor in example.items():
-        key_to_tensor_list[key].append(tensor)
-    key_to_batched_tensors = {}
-    for key in key_to_tensor_list:
-      batched_tensors = torch.stack(key_to_tensor_list[key])
-      batched_tensors = _convert_to_device(batched_tensors, self._device)
-      key_to_batched_tensors[key] = batched_tensors
-    predictions = model(**key_to_batched_tensors, **inference_args)
-    return [PredictionResult(x, y) for x, y in zip(batch, predictions)]
+    with torch.no_grad():
```

Comment:
since its not apparent why this is here. Consider linking the issue.

Reply:
https://github.com/apache/beam/issues/22811

I don't think we should link the url `https://github.com/apache/beam/issues/22811` explicitly in the comments since it would technically be resolved and simply add confusion. 

Unless you think it would be better to just incorporate this particular change in a new PR? Wouldn't be hard to do

Reply:
I'm just thinking, if a future developer came in and said. Hey why is torch.no_grad here? why do we need this check?

I wouldn't know, it wouldn't be apparent to me without a comment. But I suppose it might also be discoverable through blame.  I'll leave it up to you.

Reply:
Makes sense. Added a comment